tools/FakeItEasy.Build/Program.cs

```diff
@@ -73,9 +76,57 @@ public static void Main(string[] args)
                         }
                     });
 
+            Target(
+                "initialize-user-properties",
+                () =>
+                    {
+                        if (!File.Exists("FakeItEasy.user.props"))
+                        {
+                            var defaultUserProps = @"
+<Project>
+  <PropertyGroup>
+    <BuildProfile></BuildProfile>
+  </PropertyGroup>
+</Project>".Trim();
+                            File.WriteAllText("FakeItEasy.user.props", defaultUserProps, Encoding.UTF8);
+                        }
+                    });
+
+            foreach (var profile in Directory.EnumerateFiles("profiles", "*.props").Select(Path.GetFileNameWithoutExtension))
+            {
+                Target(
+                    "use-profile-" + profile,
+                    DependsOn("initialize-user-properties"),
+                    () =>
+                        {
+                            var xmlDoc = XDocument.Load("FakeItEasy.user.props");
+
+                            var propertyGroupElement = GetOrAddElement(xmlDoc.Root, "PropertyGroup");
+                            var buildProfileElement = GetOrAddElement(propertyGroupElement, "BuildProfile");

```

Comment:
This assumes the `BuildProfile` element is in the first `PropertyGroup`... it might not be the case. That's why I first looked for `BuildProfile` in my suggestion.

Reply:
Your error- and future-proofing is infuriating!
Infuriatingly correct and appropriate.

I've pushed a thing. It's exactly your thing, so it probably satisfies.
(Oh, as before, I'm no longer preserving whitespace, as it seemed the lesser of the two formatting evils.)

Sorry to have taken so much of your time with my whim.

Reply:
> (Oh, as before, I'm no longer preserving whitespace, as it seemed the lesser of the two formatting evils.)

👍 

> Sorry to have taken so much of your time with my whim.

You haven't!